src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -8061,8 +8054,235 @@ static class C
+            // (3,5): error CS1503: Argument 1: cannot convert from 'D' to 'System.Span<string>'
+            // C.M(new D());
+            Diagnostic(ErrorCode.ERR_BadArgType, "new D()").WithArguments("1", "D", "System.Span<string>").WithLocation(3, 5),
+            // (4,5): error CS1503: Argument 1: cannot convert from 'D' to 'System.Span<string>'
+            // C.M(new D());
+            Diagnostic(ErrorCode.ERR_BadArgType, "new D()").WithArguments("1", "D", "System.Span<string>").WithLocation(4, 5));
+
+        var expectedDiagnostics = new[]
+        {
+            // (3,5): error CS0457: Ambiguous user defined conversions 'D.implicit operator Span<object>(D)' and 'D.implicit operator ReadOnlySpan<string>(D)' when converting from 'D' to 'ReadOnlySpan<object>'
+            // C.M(new D());
+            Diagnostic(ErrorCode.ERR_AmbigUDConv, "new D()").WithArguments("D.implicit operator System.Span<object>(D)", "D.implicit operator System.ReadOnlySpan<string>(D)", "D", "System.ReadOnlySpan<object>").WithLocation(3, 5),
+            // (4,5): error CS0457: Ambiguous user defined conversions 'D.implicit operator Span<object>(D)' and 'D.implicit operator ReadOnlySpan<string>(D)' when converting from 'D' to 'ReadOnlySpan<object>'
+            // C.M(new D());
+            Diagnostic(ErrorCode.ERR_AmbigUDConv, "new D()").WithArguments("D.implicit operator System.Span<object>(D)", "D.implicit operator System.ReadOnlySpan<string>(D)", "D", "System.ReadOnlySpan<object>").WithLocation(4, 5)
+        };
+
+        CreateCompilationWithSpanAndMemoryExtensions(source, parseOptions: TestOptions.RegularNext).VerifyDiagnostics(expectedDiagnostics);
+        CreateCompilationWithSpanAndMemoryExtensions(source).VerifyDiagnostics(expectedDiagnostics);
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void OverloadResolution_SpanVsReadOnlySpan_07(LanguageVersion langVersion)
+    {
+        var source = """
+            using System;
+            
+            C.M(new D());
+            
+            static class C
+            {
+                public static void M(Span<string> arg) => Console.Write(1);
+                public static void M(ReadOnlySpan<object> arg) => Console.Write(2);
+            }
+            
+            class D
+            {
+                public static implicit operator Span<string>(D d) => default;
+                public static implicit operator ReadOnlySpan<object>(D d) => default;
+            }
+            """;
+        CreateCompilationWithSpanAndMemoryExtensions(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion)).VerifyDiagnostics(
+            // (3,3): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(Span<string>)' and 'C.M(ReadOnlySpan<object>)'
+            // C.M(new D());
+            Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(System.Span<string>)", "C.M(System.ReadOnlySpan<object>)").WithLocation(3, 3));
+    }
```

Comment:
>} [](http://example.com/codeflow?start=4&length=1)

Should we also test `Span<object>` and `ReadOnlySpan<object>`:
```csharp
C.M(new D());
            
static class C
{
    public static void M(Span<object> arg) => Console.Write(1);
    public static void M(ReadOnlySpan<object> arg) => Console.Write(2);
}
            
class D
{
    public static implicit operator Span<object>(D d) => default;
    public static implicit operator ReadOnlySpan<object>(D d) => default;
}
```

And `Span<object>` and `ReadOnlySpan<string>`:
```csharp
C.M(new D());
            
static class C
{
    public static void M(Span<object> arg) => Console.Write(1);
    public static void M(ReadOnlySpan<string> arg) => Console.Write(2);
}
            
class D
{
    public static implicit operator Span<object>(D d) => default;
    public static implicit operator ReadOnlySpan<string>(D d) => default;
}
```